napari/layers/base/base.py

```diff
@@ -998,6 +1008,132 @@ def _data_to_world(self) -> Affine:
         """
         return self._transforms[1:3].simplified
 
+    def vector_world_to_data(self, vector) -> tuple:
+        """Convert a vector defining an orientation from world coordinates to data coordinates.
+        For example, this would be used to convert the view ray.
+
+        Parameters
+        ----------
+        vector : tuple, list, 1D array
+            A vector in world coordinates.
+
+        Returns
+        -------
+        tuple
+            Vector in data coordinates.
+        """
+        vector = np.asarray(vector)
+
+        # create transform for view direction (world -> layer data)
+        shear_matrix = expand_upper_triangular(self.shear)
+        rot_matrix = self.rotate
+        scale_matrix = np.diag(self.scale)
+        world_to_layer = np.linalg.inv(
+            shear_matrix @ rot_matrix
+        ).T @ np.linalg.inv(scale_matrix)
+
+        return tuple(vector @ world_to_layer)
+
+    @property
+    def _display_bounding_box(self):
+        """An axis aligned (self._ndisplay, 2) bounding box around the data"""
+        return self._extent_data[:, self._dims_displayed].T
+
+    def _data_face_intersection(
```

Comment:
@alisterburt , I think I have updated this so the bounding box intersection detection is performed in world coordinates (i.e., we no longer need the `canvas_to_world` transform on the layer). Would you mind taking a look and letting me know what you think?

Reply:
hahah I'm literally working on the same thing, will look at yours now!

Reply:
oh wait, not working on the same thing - I was rejigging the inside triangles bit

Reply:
The basic approach is:
1. convert the data bounding box face vertices into world coordinates
2. project the bounding box face vertices onto the plane described by the click point and the view normal direction 
3. now that the now that the bounding box face vertices and click point are in the same plane, I rotate them so the plan is axis aligned (i.e., normal vector [0, 0, 1])
4. I take the 2D coordinates (coords for axes 0 and 1) of the face vertices and click point and perform our standard "click in 2D triangle" detection.

Reply:
> hahah I'm literally working on the same thing, will look at yours now!

Dang! I hope we didn't duplicate too much work, haha. Let me know what you think!

Reply:
Update: chatted with @alisterburt and we will keep this approach. I have slightly updated it so that we now do all of the comparisons in data coordinates.